src/Chapter2.hs

```diff
@@ -857,7 +889,9 @@ and reverses it.
   function, but in this task, you need to implement it manually. No
   cheating!
 -}
-rewind = error "rewind: Not Implemented!"
+-- rewind [] = []
+rewind :: [a] -> [a]
+rewind = foldl (flip (:)) []
```

Comment:
Wow, that's a clever solution using standard functions! 🍸 

And you correctly noticed that you don't need to handle the case with `[]` 🙂